design/nab_status_update.md

```diff
@@ -0,0 +1,82 @@
+# Developer Workflow: NonAdminBackup Phase and Conditions within NonAdminBackup Status
+
+## Overview
+
+This document outlines the design around updating NonAdminBackup objects Phase and Conditions within Status.
+
+### Phase
+
+A NonAdminBackup's `status` field has a `phase` field, which is updated by NAC controller.
+
+The `phase` is a simple one high-level summary of the lifecycle of an NonAdminBackup.
+
+It is always a one well defined value, that is intended to be a comprehensive state of a NonAdminBackup.
+
+Those are are the possible values for phase:
+
+| **Value** | **Description**                 |
+|-----------|--------------------------------|
+| New | *NonAdminBackup resource was accepted by the OpenShift cluster, but it has not yet been processed by the NonAdminController* |
+| BackingOff | Velero *Backup* object was not created due to NonAdminBackup error (configuration or similar) |
+| Created | Velero *Backup* was created. The Phase will not have additional informations about the |
+
+### Conditions
+
+The `conditions` is also a part of the NonAdminBackup's `status` field. One NAB object may have multiple conditions. It is more granular knowledge of the NonAdminBackup object and represents the array of the conditions through which the NonAdminBackup has or has not passed. Each `NonAdminCondition` has one of the following `type`:
+
+| **Condition** | **Description**                 |
```

Comment:
@mpryc I think we also need a `Error` condition here. So that whenever the backup is in `BackingOff` phase the `Error` condition is applied to the NAB object

Reply:
When the `BackingOff` we set the `BackupAccepted` condition to `False` see the diagram in another comment. Do we need to have Error condition as well? I think it's a bit too many states.

The `BackupAccepted` set to False have also the string reason/message where we can say why it was wrong.

Reply:
Example:
```yaml
status:
  conditions:
    - lastTransitionTime: '2024-04-17T20:56:57Z'
      message: NonAdminBackup CR does not contain valid BackupSpec
      reason: invalid_backupspec
      status: 'False'
      type: BackupAccepted
  phase: BackingOff
```

Reply:
When the user sees an `Error` condition they definitely think that something is not working as intended/expected.
The word `Accepted` does not imply the same. I agree that the `Accepted` condition can display the error message but having an `Error` condition would improve the user experience.

Reply:
We have `phase: BackingOff` for that, so this is high level information to tell that something went wrong.

User has the `Accepted: False` status, of course we could add `Accepted: False` and `Error: True` or just `Error: True`, but that in my opinion is duplication of what phase is for.